### PR TITLE
Refactor initialize groups tasks used for byo

### PR DESCRIPTION
**Pull request by mtnbikenc**
Two tasks for initializing group names for the byo playbooks were located
in the common folder in the std_include.yml file.  Byo dependencies
should not be in the common folder.  The two tasks have been removed
from common/openshift-cluster/std_include.yml to a new file
byo/openshift-cluster/initialize_groups.yml.  All references where these
tasks were included from either std_include.yml or other various files
have been updated to use the byo initialize_groups.yml.  The methodology
implemented follows the pattern of having groups set up in byo then
calling out to playbooks in common, which are common to all deployments.

Entry point playbooks identified/modified in this change:
```
$ grep -ri --include=\*.{yml,yaml} -E 'initialize_groups\.yml'
playbooks/byo/openshift-cluster/redeploy-registry-certificates.yml:- include: initialize_groups.yml
playbooks/byo/openshift-cluster/openshift-logging.yml:- include: initialize_groups.yml
playbooks/byo/openshift-cluster/redeploy-certificates.yml:- include: initialize_groups.yml
playbooks/byo/openshift-cluster/enable_dnsmasq.yml:- include: initialize_groups.yml
playbooks/byo/openshift-cluster/redeploy-master-certificates.yml:- include: initialize_groups.yml
playbooks/byo/openshift-cluster/redeploy-router-certificates.yml:- include: initialize_groups.yml
playbooks/byo/openshift-cluster/redeploy-etcd-certificates.yml:- include: initialize_groups.yml
playbooks/byo/openshift-cluster/config.yml:- include: initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/docker/upgrade.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_6/upgrade_nodes.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_6/upgrade.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_6/upgrade_control_plane.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_3/upgrade_nodes.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_3/upgrade.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_3/upgrade_control_plane.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_5/upgrade_nodes.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_5/upgrade.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_5/upgrade_control_plane.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_4/upgrade_nodes.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_4/upgrade.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/v3_4/upgrade_control_plane.yml:- include: ../../initialize_groups.yml
playbooks/byo/openshift-cluster/upgrades/upgrade_etcd.yml:- include: ../initialize_groups.yml
playbooks/byo/openshift-cluster/redeploy-openshift-ca.yml:- include: initialize_groups.yml
playbooks/byo/openshift-cluster/redeploy-node-certificates.yml:- include: initialize_groups.yml
playbooks/byo/openshift-node/scaleup.yml:- include: ../openshift-cluster/initialize_groups.yml
playbooks/byo/openshift-node/restart.yml:- include: ../openshift-cluster/initialize_groups.yml
playbooks/byo/openshift-master/scaleup.yml:- include: ../openshift-cluster/initialize_groups.yml
playbooks/byo/openshift-master/restart.yml:- include: ../openshift-cluster/initialize_groups.yml
playbooks/byo/openshift_facts.yml:- include: openshift-cluster/initialize_groups.yml
playbooks/byo/rhel_subscribe.yml:- include: openshift-cluster/initialize_groups.yml
playbooks/byo/openshift-etcd/restart.yml:- include: ../openshift-cluster/initialize_groups.yml
```

### CODE REVIEW

**Comment by mtnbikenc** [comment]
aos-ci-test

File: playbooks/byo/openshift-cluster/initialize_groups.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  become: no
+  gather_facts: no
+  tags:
+  - always
+  tasks:
+  - include_vars: cluster_hosts.yml
+  - name: Evaluate group l_oo_all_hosts
+    add_host:
+      name: "{{ item }}"
+      groups: l_oo_all_hosts
```
Comment (kwoodson) [inline playbooks/byo/openshift-cluster/initialize_groups.yml]: What's the significance of putting `l_` on this variable?  This is for my own sake.  I've seen this pattern used else where in this repo.

I generally shy away from naming variables with `l` as it appears to the viewer as `l`, `L`, `|`, or even `1`.

Comment (sdodson) [inline playbooks/byo/openshift-cluster/initialize_groups.yml]: local, something internal to the role that one should never expect to set from the outside

Comment (kwoodson) [inline playbooks/byo/openshift-cluster/initialize_groups.yml]: @sdodson, thanks for the explanation.  The irony of it is, if its internal to a role, then why are we in a playbook defining it.  At that point it becomes global.   Its fine for now but I'd prefer if we moved away from `l_`.  I'd prefer `group_`  or something that denotes our intention.

Comment (mtnbikenc) [inline playbooks/byo/openshift-cluster/initialize_groups.yml]: @kwoodson From what I've been able to dig out, the `l_oo_all_hosts` group was supposed to be something that was only referenced in the original playbook for creating the all `g_` group names, which are then processed into the `oo_` group names.  The group name initialization process is something I have a card for to come back to.  I've found where this `l_oo_all_hosts` group name has been used in other places, outside of the original playbook (probably by mistake), which should be corrected.

Comment (kwoodson) [inline playbooks/byo/openshift-cluster/initialize_groups.yml]: @sdodson @mtnbikenc, Thanks for the explanations.

**Comment by mtnbikenc** [comment]
aos-ci-test

**Comment by mtnbikenc** [comment]
[merge]

**Comment by kwoodson** [comment]
aos-ci-test

**Comment by kwoodson** [comment]
flake https://github.com/openshift/origin/issues/13271

**Comment by kwoodson** [comment]
aos-ci-test

**Comment by kwoodson** [comment]
[merge]

**Comment by mtnbikenc** [comment]
Flake https://github.com/openshift/origin/issues/13108

**Comment by mtnbikenc** [comment]
Last test failed because of Ansible 2.3 being used for unit testing.  Fixed in #3915.  Trying again.

[merge]

**Comment by mtnbikenc** [comment]
@sdodson can you re-trigger the test?

**Comment by mtnbikenc** [comment]
aos-ci-test

**Pull request merged**